### PR TITLE
grabbag of doctests

### PR DESCRIPTION
**Pull request by KristofferC**

### CODE REVIEW

File: base/math.jl [hunk 1]
```diff
@@ -436,6 +436,19 @@ sqrt(x::Real) = sqrt(float(x))
     hypot(x, y)
 
 Compute the hypotenuse ``\\sqrt{x^2+y^2}`` avoiding overflow and underflow.
+
+```
```
Comment (fredrikekre) [inline base/math.jl]: `jldoctest`

File: base/sort.jl [hunk 2]
```diff
@@ -553,6 +620,22 @@ end
 
 Like [`sortperm`](@ref), but accepts a preallocated index vector `ix`.  If `initialized` is `false`
 (the default), ix is initialized to contain the values `1:length(v)`.
```
Comment (tkelman) [inline base/sort.jl]: `ix` should be code highlighted

Comment (KristofferC) [inline base/sort.jl]: done

File: base/sort.jl [hunk 3]
```diff
@@ -432,6 +443,30 @@ providing a custom "less than" function; use `rev=true` to reverse the sorting o
...
 `rev=true` reverses whatever ordering specified via the `by` and `lt` keywords.
+
+```jldoctest
+julia> v = [3, 2, 1]; sort!(v); v
+3-element Array{Int64,1}:
+ 1
+ 2
+ 3
+
+julia> v = [3, 2, 1]; sort(v, rev = true); v
```
Comment (KristofferC) [inline base/sort.jl]: should be `sort!` here, and also probably shouldn't arrange `v` so that `rev = true` version is a noop

Comment (tkelman) [inline base/sort.jl]: the no-op part of this looks like it wasn't addressed

Comment (KristofferC) [inline base/sort.jl]: Thanks, fixed

**Pull request merged**